### DISCUSSION TITLE
Sensor68 values for ELTAKO WS anemometer

### DISCUSSION THREAD

**Discussion by Jambola77** [arendst/Tasmota#14660] [Q&A] [answered]
Hello,
does anyone has the Sensor68 values for the ELTAKO WS anemometer?
I have so far:
Sensor68 1, 51 (radius from center to edge is 51.5mm)
Sensor68 2, 2 (two impulses for one turnaround)

But what is speed factor (Sensor68 4)? 

The wind speed formula for the ELTAKO is: v [km/h] = 1.761 / (1 + rps) + 3.013 * rps

**Comment by barbudor** [comment, marked as answer]
This is the formula in the code:
```
  speed = RPS * 2 * Pi * (radius / 1000) * speed_factor
```
Apparently there's no way to apply the formula as specified by Eltako.

So my guess at 1st approximation would be speed_factor = (3,013 * 1000)/(2 * pi * 51) = 9,4
However it will probably less accurate at low speed

**Reply by sfromis** [reply]
If you want to use the formula from Eltako directly, you could fudge the Sensor68 values to cancel themselves out, returning the "raw" RPS value, and then use rules (or whatever) to do the calculation based on the RPS value.

Something like radius 159.155 (1000 / Pi / 2) and speed factor 1 would work to come very close the the original RPS, assuming that you can use decimals. Of course, you could "distribute" the 159.155 among radius and speed factor, if that's useful somehow.

**Reply by Jambola77** [reply]
Maybe someone can add this to the line 125 of xsns_68_windmeter.ino and add the #define USE_WINDMETER_ELTAKO to the config file:

```
#ifndef USE_WINDMETER_ELTAKO
  WindMeter.speed = ((WindMeter.counter / Settings.windmeter_pulses_x_rot) * (windmeter_2pi * ((float)Settings.windmeter_radius / 1000))) * ((float)Settings.windmeter_speed_factor / 1000);
  #else 
//use below for ELTAKO WS anemometer
  // formula is  v [km/h] = 1.761 / (1 + rps) + 3.013 * rps
  WindMeter.speed = (float)(1.761 / (1 + WindMeter.counter) + 3.013 * WindMeter.counter);  
  #endif
```

**Reply by barbudor** [reply]
You're welcome to propose a PR
But however your formula is wrong as `counter` is not `rps`

**Comment by Jambola77** [comment]
I can see these formula:
 speed = RPS * 2 * Pi * (radius / 1000) * speed_factor
which leads in the code to:
WindMeter.speed = ((WindMeter.counter / Settings.windmeter_pulses_x_rot) * (windmeter_2pi * ((float)Settings.windmeter_radius / 1000))) * ((float)Settings.windmeter_speed_factor / 1000);

So, I assume, that RPS= WindMeter.counter / Settings.windmeter_pulses_x_rot

Then the formula for Eltako (2 pulses for 1 round) is:
 WindMeter.speed = (float)(1.761 / (1 + (WindMeter.counter / 2)) + 3.013 * (WindMeter.counter / 2));
